charts/newrelic-infrastructure-v3/templates/control-plane/daemonset.yaml

```diff
@@ -192,10 +192,20 @@ spec:
       {{- with .Values.priorityClassName }}
       priorityClassName: {{ . }}
       {{- end }}
-      {{- with .Values.controlPlane.affinity.nodeAffinity }}
+      {{- if .Values.controlPlane.affinity.nodeAffinity }}
       affinity:
         nodeAffinity:
-          {{- . | toYaml | nindent 10 }}
+          {{- .Values.controlPlane.affinity.nodeAffinity | toYaml | nindent 10 }}
+      {{- else if include "newrelic.fargate" . }}
```

Comment:
Is it possible to scrape the EKS control plane on Fargate? If it isn't, we could get rid of the fargate logic here.